File: frontend/src/lib/api/history.ts
/**
 * History / anniversary endpoints.
 *
 * These power the "on this day" niche: a curated database of notable events that
 * can seed a project idea without any external call.
 */

import * as api from "@/lib/api/client";
import { OnThisDayEvent } from "@/types/history";

/** ``GET /history/on-this-day`` — today's events, or a specific date's. */
export function onThisDay(params: {
  month?: number;
  day?: number;
} = {}): Promise<OnThisDayEvent[]> {
  return api.apiFetch<OnThisDayEvent[]>("/history/on-this-day", {
    query: params,
  });
}

/** ``GET /history/search`` — search the event database by keyword or location. */
export function searchHistory(q: string): Promise<OnThisDayEvent[]> {
  return api.apiFetch<OnThisDayEvent[]>("/history/search", { query: { q } });
}

File: src/content_factory/api/routers/history.py
"""Historical events, maps and infographics."""

from __future__ import annotations

from fastapi import APIRouter

from ...models import OnThisDayEvent
from ...service import ContentFactoryService


def build_router(service: ContentFactoryService) -> APIRouter:

    router = APIRouter()

    @router.get("/history/on-this-day", response_model=list[OnThisDayEvent])
    def get_on_this_day(
        month: int | None = None, day: int | None = None
    ) -> list[OnThisDayEvent]:
        """Get notable historical disasters/events for today or a specific date."""
        return service.get_on_this_day(month=month, day=day)

    @router.get("/history/search", response_model=list[OnThisDayEvent])
    def search_history(q: str) -> list[OnThisDayEvent]:
        """Search historical catastrophe database by keyword or location."""
        return service.search_history_events(q)

    return router
